projects/netgrif-components-core/src/lib/data-fields/case-ref-field/abstract-case-ref-field.component.ts

```diff
@@ -0,0 +1,250 @@
+export abstract class AbstractCaseRefFieldComponent implements AfterViewInit {
+
+    @Input() public dataField: CaseRefField;
+
+    constructor(protected _petriflowCanvasService: PetriflowCanvasService, protected _petriflowFactoryService: PetriflowCanvasFactoryService,
+                protected _petriflowConfigService: PetriflowCanvasConfigurationService, protected _caseResourceService: CaseResourceService,
+                protected _petriNetResourceService: PetriNetResourceService, protected _log: LoggerService, protected _snackBar: SnackBarService,
+                protected _translate: TranslateService) {
+    }
+
+    ngAfterViewInit(): void {
+        this._petriNetResourceService.getNetByCaseId(this.dataField.value).subscribe(net => {
+            if (net) {
+                this.createNet(net);
+            }
+        }, error => {
+            this._log.error('Getting net by Case ID failed in field ['+ this.dataField.stringId + ']', error);
+            this._snackBar.openErrorSnackBar(this._translate.instant('dataField.snackBar.caseNetGetFailed'));
+        });
+    }
+
+    protected createNet(net: PetriNetImport) {
+        const trans: Array<PetriflowTransition> = [];
+        const places: Array<PetriflowPlace> = [];
+        const arcs: Array<PetriflowArc<any>> = [];
+        let minX: number = Number.MAX_SAFE_INTEGER;
+        let minY: number = Number.MAX_SAFE_INTEGER;
+        net.transitions.forEach((value) => {
+            const t = this.createTransitions(value)
+            trans.push(t);
+            minX = Math.min(minX, value.position.x);
+            minY = Math.min(minY, value.position.y);
+            this.setPlaceActions(t.canvasElement.element);
+        })
+        net.places.forEach((value) => {
+            const p = this.createPlace(value)
+            places.push(p);
+            minX = Math.min(minX, value.position.x);
+            minY = Math.min(minY, value.position.y);
+            this.setPlaceActions(p.canvasElement.element);
+            p.canvasElement.markingTokens.forEach(markingToken => {
+                this.setPlaceActions(markingToken);
+            });
+        })
+        net.arcs.forEach((arc) => {
+            const a = this.createArcs(trans, places, arc, net)
+            arcs.push(a);
+            arc.breakpoints?.forEach(value => {
+                minX = Math.min(minX, value.x);
+                minY = Math.min(minY, value.y);
+            });
+            this.setPlaceActions(a.element.arcLine);
+        });
+        trans.forEach(value => {
+            if (net.assignedTasks.includes(value.canvasElement.label.textContent)) {
+                value.select();
+            }
+            if (this.isEnabled(value, places, arcs)) {
+                value.canvasElement.element.classList.add('svg-transition-enabled');
+                value.canvasElement.element.setAttributeNS(null, 'fill', 'yellowgreen');
+                value.canvasElement.element.setAttributeNS(null, 'stroke', 'green');
```

Comment:
should the colors not be customisable / loaded from a css pallete?

Reply:
no, only css classes